public/javascript/posts/BookAppointmentModal.js

```diff
@@ -149,7 +161,12 @@ angular.module('BookAppointmentApp')
       },
 
       paymentSelected: function(){
-        return true;
+        if(selected_event.prepay_amount > 0){
```

Comment:
Instead of if .. else you can do 

`return selected_event.prepay_amount > 0 ? true : false`

also will only `selected_event` work , should this not be `$scope.selected_event`